File: CSharp/LINQtoDB/Models/Person.cs
using LinqToDB;
using LinqToDB.Mapping;

namespace Models;

[Table("people")]
public class Person
{
    [PrimaryKey, Column("id"), DataType(dataType: DataType.Text)]
    public Guid Id { get; set; }

    [Column("name"), NotNull]
    public string Name { get; set; } = "";

    [Column("address"), NotNull]
    public string Address { get; set; } = "";

    [Column("birth_date"), NotNull]
    public DateOnly BirthDate { get; set; }

    public Person()
    {
        Id = Guid.NewGuid();
    }

    public override string ToString() => $"{{ Person: {{ Id: {Id}, Name: {Name}, Address: {Address}, BirthDate: {BirthDate:yyyy-MM-dd} }} }}";
}

File: FSharp/LINQtoDB/Models/Person.fs
namespace Models

open LinqToDB
open LinqToDB.Mapping
open System

[<Table "people">]
type Person (name, address, birthDate) =
  new () =
    let today = DateTime.Today
    Person ("", "", DateOnly (today.Year, today.Month, today.Day))

  [<PrimaryKey>][<Column "id">][<DataType (dataType=DataType.Text)>]
  member val Id = Guid.NewGuid () with get, set

  [<Column "name">][<NotNull>]
  member val Name = name with get, set

  [<Column "address">][<NotNull>]
  member val Address = address with get, set

  [<Column "birth_date">][<NotNull>]
  member val BirthDate = birthDate with get, set

  with override this.ToString () =
        let format = "yyyy-MM-dd"
        $"{{ Person: {{ Id: {this.Id}, Name: {this.Name}, Address: {this.Address}, BirthDate: {this.BirthDate.ToString format} }} }}"
